etcdserver/server.go

```diff
@@ -1988,6 +1992,62 @@ func (s *EtcdServer) sync(timeout time.Duration) {
 	})
 }
 
+// publishV3 registers server information into the cluster using v3 request. The
+// information is the JSON representation of this server's member struct, updated
+// with the static clientURLs of the server.
+// The function keeps attempting to register until it succeeds,
+// or its server is stopped.
+// TODO: replace publish() in 3.6
+func (s *EtcdServer) publishV3(timeout time.Duration) {
+	req := &membershippb.ClusterMemberAttrSetRequest{
+		Member_ID: uint64(s.id),
+		MemberAttributes: &membershippb.Attributes{
+			Name:       s.attributes.Name,
+			ClientUrls: s.attributes.ClientURLs,
+		},
+	}
+	lg := zap.NewNop()
```

Comment:
I meant, we still use the zap logger, but a separate function `getLogger` should handle the previous if-statements. Can we use the logger from `EtcdServer` in this PR, and iterate more to clean up `getLogger` method in the separate PR?

Reply:
Got you. Will update `getLogger()` in another PR.